src/test/java/org/folio/rest/impl/UsersAPIIT.java

```diff
@@ -1,99 +1,120 @@
 import org.folio.support.User;
-import org.junit.Assert;
-import org.junit.jupiter.api.AfterAll;
+import org.folio.support.ValidationErrors;
+import org.folio.support.VertxModule;
+import org.folio.support.http.FakeTokenGenerator;
+import org.folio.support.http.GroupsClient;
+import org.folio.support.http.OkapiHeaders;
+import org.folio.support.http.UsersClient;
 import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.testcontainers.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.extension.ExtendWith;
 
-import io.restassured.RestAssured;
-import io.restassured.http.ContentType;
-import io.restassured.response.Response;
-import io.restassured.specification.RequestSpecification;
-import io.vertx.core.DeploymentOptions;
 import io.vertx.core.Vertx;
-import io.vertx.core.json.JsonObject;
+import io.vertx.junit5.Timeout;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
 import lombok.SneakyThrows;
 
 /**
  * Most old UsersAPI tests are in deprecated org.folio.moduserstest.RestVerticleIT and
  * should be moved here.
  */
+@Timeout(value = 20, timeUnit = SECONDS)
+@ExtendWith(VertxExtension.class)
 class UsersAPIIT {
   static final String HOME_ADDRESS_TYPE_ID = "93d3d88d-499b-45d0-9bc7-ac73c3a19880";
   static final String ClAIM_ADDRESS_TYPE_ID = "b6f4d1c6-0dfa-463c-9534-f49c4f0ae090";
-  static final String TENANT = "usersapiit";
-  static final String TOKEN = "header." + Base64.getEncoder().encodeToString("{}".getBytes()) + ".signature";
-  static Vertx vertx;
-  static String baseUrl;
+  private static UsersClient usersClient;
+  private static GroupsClient groupsClient;
 
   @BeforeAll
-  static void beforeAll() {
-    vertx = Vertx.vertx();
+  @SneakyThrows
+  static void beforeAll(Vertx vertx, VertxTestContext context) {
+    final var tenant = "users_integration_tests";
+    final var token = new FakeTokenGenerator().generateToken();
 
     PostgresClient.setPostgresTester(new PostgresTesterContainer());
 
-    RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
-    RestAssured.port = NetworkUtils.nextFreePort();
-    baseUrl = "http://localhost:" + RestAssured.port;
-    deploy();
-    deleteTenantIgnore();  // remove left over from previous test run
-    postTenant();
-  }
+    final var port = NetworkUtils.nextFreePort();
+
+    final var headers = new OkapiHeaders("http://localhost:" + port,
+      tenant, token);
+
+    usersClient = new UsersClient(new URI("http://localhost:" + port), headers);
+    groupsClient = new GroupsClient(new URI("http://localhost:" + port), headers);
+
+    final var module = new VertxModule(vertx);
 
-  @AfterAll
-  static void afterAll() {
-    vertx.close();
+    module.deployModule(port)
+      .onComplete(context.succeeding(res -> module.enableModule(headers,
+          true, false)
```

Comment:
This is the only place where enableModule loads reference data. Perhaps another test or this one to also load sample data and check that *something* is loaded.

Reply:
Agreed. I'll add a test specifically for loading the sample data.

The existing tests only really check the sample data very superficially or accidentally.